src/OrchardCore.Modules/Orchard.Autoroute/Handlers/AutoroutePartHandler.cs

```diff
@@ -119,5 +124,29 @@ private string GetPattern(AutoroutePart part)
 
             return pattern;
         }
+
+        private string GenerateUniquePath(string path, AutoroutePart context)
```

Comment:
I have found another problem with the validation of generated path in `AutoroutePathHandler` ... I am unable to find a proper way to present the validation error to user.

How it works:
* user saves a content item
* `AutoroutePartDisplayDriver.UpdateAsync` is called. It validates user input and creates a `Shape` from the current values of the `AutoroutePart`
* `AutoroutePartHandler.Updated` is called. If `Path` is empty, it generates a new one.
* Path should be validated for unicity, if it isn't unique I can add a model error (`IUpdater` is available in the handler) but I can't update the `Shape` generated by the driver
* Result: User sees the validation error, but not the generated `Path`

(If there isn't any validation error everything works fine, because at the end user is redirected to the `Edit` action and the original shape isn't used)

I don't see any solution short of my original unique path generator, which will also work for programmatic updates of the content item.

Dou you have any other suggestions?